### PR TITLE
improve evaluators

### PR DESCRIPTION
**Pull request by MaHaWo**
This PR carries over evaluator improvements from the refactor-evaluators branch. It introduces a more flexible version of the setup and allows for better checkpoint reconstruction in the future. 

- augment evaluator and tests
- update exports
- update trainer and tests
- remove juliacall and juliaworker because of unreliability and slowness

### CODE REVIEW

File: QuantumGravPy/src/QuantumGrav/evaluate.py [hunk 1]
```diff
@@ -1,168 +1,256 @@
...
+            for i, batch in enumerate(tqdm.tqdm(data_loader, desc="Evaluating")):
                 data = batch.to(self.device)
                 if self.apply_model:
                     outputs = self.apply_model(model, data)
                 else:
                     outputs = model(data.x, data.edge_index, data.batch)
                 loss = self.criterion(outputs, data)
-                current_data.append(loss)
+                current_losses.append(loss.unsqueeze(0))
+                current_predictions.append(outputs)
```
Comment (causal-wagner) [inline QuantumGravPy/src/QuantumGrav/evaluate.py:132]: If we save all predictions, say in a validation set, can that becomes memory-heavy? I'm thinking now about node-level predictions like the curvature in a point.

File: QuantumGravPy/src/QuantumGrav/evaluate.py [hunk 2]
```diff
@@ -1,168 +1,256 @@
...
-            for i, batch in enumerate(data_loader):
+            for i, batch in enumerate(tqdm.tqdm(data_loader, desc="Evaluating")):
                 data = batch.to(self.device)
                 if self.apply_model:
                     outputs = self.apply_model(model, data)
                 else:
                     outputs = model(data.x, data.edge_index, data.batch)
                 loss = self.criterion(outputs, data)
-                current_data.append(loss)
+                current_losses.append(loss.unsqueeze(0))
```
Comment (causal-wagner) [inline QuantumGravPy/src/QuantumGrav/evaluate.py:131]: Is loss always a scalar per batch?

**Review by causal-wagner — COMMENTED** [review]
Looks good to me. Just two small questions

File: QuantumGravPy/src/QuantumGrav/evaluate.py (hunk 2 above)
Comment (MaHaWo) [reply QuantumGravPy/src/QuantumGrav/evaluate.py:131]: yes. ultimately you need to reduce it to a scalar for backprop.

File: QuantumGravPy/src/QuantumGrav/evaluate.py (hunk 1 above)
Comment (MaHaWo) [reply QuantumGravPy/src/QuantumGrav/evaluate.py:132]: eventually yes. unfortunately, most quality measures are greedy, so we need them in one go.

Comment (causal-wagner) [reply QuantumGravPy/src/QuantumGrav/evaluate.py:132]: Ok so nothing one can do about that then...
